test/e2e/apimachinery/watch.go

```diff
@@ -377,6 +387,55 @@ var _ = SIGDescribe("Watchers", func() {
 		}
 		<-donec
 	})
+
+	framework.ConformanceIt("should receive add, update and delete notifications on namespace with legacy watch api", func(ctx context.Context) {
```

Comment:
I think these would be better as integration tests

See test server setup in TestWatchRestartsIfTimeoutNotReached in test/integration/apimachinery/watch_restart_test.go

You can start a single test server and client, then in subtests, make per-test namespaces and exercise watching them

You can iterate on the tests by starting `etcd` locally and running something like `go test ./test/integration/apimachinery -run TestYourTestName`

Reply:
OK